kernel/src/table_properties/deserialize.rs

```diff
@@ -0,0 +1,287 @@
+    }
+}
+
+impl<'de> Deserializer<'de> for StringMapDeserializer<'de> {
+    type Error = de::value::Error;
+
+    fn deserialize_any<V>(self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: Visitor<'de>,
+    {
+        visitor.visit_map(HashMapMapAccess {
+            iter: self.iter,
+            value: None,
+        })
+    }
+
+    serde::forward_to_deserialize_any! {
+        bool i8 i16 i32 i64 u8 u16 u32 u64 f32 f64 char str string bytes
+        byte_buf option unit unit_struct newtype_struct seq tuple tuple_struct
+        map struct enum identifier ignored_any
+    }
+}
+
+struct HashMapMapAccess<'de> {
+    iter: std::collections::hash_map::Iter<'de, String, String>,
+    value: Option<&'de String>,
+}
+
+impl<'de> MapAccess<'de> for HashMapMapAccess<'de> {
+    type Error = de::value::Error;
+
+    fn next_key_seed<K>(&mut self, seed: K) -> Result<Option<K::Value>, Self::Error>
+    where
+        K: DeserializeSeed<'de>,
+    {
+        match self.iter.next() {
+            Some((key, value)) => {
+                self.value = Some(value);
+                let de = de::value::StrDeserializer::new(key);
+                seed.deserialize(de).map(Some)
+            }
+            None => Ok(None),
+        }
+    }
+
+    fn next_value_seed<V>(&mut self, seed: V) -> Result<V::Value, Self::Error>
+    where
+        V: DeserializeSeed<'de>,
+    {
+        let value = self.value.take().unwrap(); // FIXME
+        let de = de::value::StrDeserializer::new(value);
+        seed.deserialize(de)
+    }
+}
+
+pub(crate) fn deserialize_option<'de, D, T>(deserializer: D) -> Result<Option<T>, D::Error>
+where
+    D: Deserializer<'de>,
+    T: Deserialize<'de>,
+{
+    T::deserialize(deserializer).map(Some)
+}
+
+pub(crate) fn deserialize_pos_int<'de, D>(deserializer: D) -> Result<Option<u64>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s: String = Deserialize::deserialize(deserializer)?;
+    let n: u64 = s.parse().unwrap(); // FIXME
+    if n == 0 {
+        panic!("FIXME");
+        // return Err("something");
+    }
+    Ok(Some(n))
+}
+
+pub(crate) fn deserialize_bool<'de, D>(deserializer: D) -> Result<Option<bool>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s: String = Deserialize::deserialize(deserializer)?;
+    match s.as_str() {
+        "true" => Ok(Some(true)),
+        "false" => Ok(Some(false)),
+        _ => Err(de::Error::unknown_variant(&s, &["true", "false"])),
+    }
+}
+
+pub(crate) fn deserialize_column_names<'de, D>(
+    deserializer: D,
+) -> Result<Option<Vec<ColumnName>>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s: String = Deserialize::deserialize(deserializer)?;
+    Ok(Some(
+        s.split(',')
+            .map(|name: &str| ColumnName::new([name]))
+            .collect(),
+    ))
+}
+
+pub(crate) fn deserialize_interval<'de, D>(deserializer: D) -> Result<Option<Duration>, D::Error>
+where
+    D: Deserializer<'de>,
+{
```

Comment:
Another spot to potentially use the alternative syntax:
```suggestion
pub(crate) fn deserialize_interval<'de, D: Deserializer<'de>>(
    deserializer: D,
) -> Result<Option<Duration>, D::Error> {
```
(same considerations as before)

Reply:
would cause a line split so I'll just leave it!

Reply:
what line split, sorry?

Reply:
Oh i see - yours already has line split. do you think the non-where-clause is cleaner to read? i guess it does save a line..